### ISSUE TITLE
SA 1642 Is Triggered with Private Constructor Text.

### ISSUE THREAD

**Issue by lumberjack4** [DotNetAnalyzers/StyleCopAnalyzers#1318]
```
/// <summary>
/// Prevents a default instance of the <see cref="MyClass" /> class from being created externally.
/// </summary>
private MyClass()
{
}
```

Triggers SA 1642.  Code fix attempts to replace the summary with the public constructor default text.  This was seen using beta 8.

**Comment by sharwell** [comment]
You are seeing two different things here.

1. SA1642 is being reported because you didn't end your sentence following the word `created`. The analyzer currently expected `created.` (with the `.`).

2. The replacement text is an intentional deviation from StyleCop "classic" (#413). SA1642 supports *either* wording for private constructors, and the code fix prefers the wording you saw being inserted.

My recommendation is updating your constructors to use the non-misleading public constructor wording.

@pdelvo Should we drop the `.` from the expected text in the analyzer?

**Comment by lumberjack4** [comment]
I like the idea of leaving off the trailing `.`   This appears to be the behavior for the normal public constructor text as most of my constructor documentation has more than just the starting text sequence and none of them trigger the diagnostic.

**Comment by sharwell** [comment]
Keep in mind that your current constructor text is misleading. From outside the class, the constructor is not visible at all, which means it's not showing up in IntelliSense and you can't see the summary. The only place where the summary is visible is within the context of your current class, and in that scope it does not prevent the creation of new instances of the class.

* If you want to prevent creation of instances of a class, use a `static` class.
* If you
